### PR TITLE
Added Intents to make it easier when passing intents to client

### PR DESCRIPTION
**Pull request by an-dyy**
Tested on my side, working fine currently. Let me know If I need to change anything.

### CODE REVIEW

**Comment by an-dyy** [comment]
New commit allows for Intents to behave correctly.
```py
intents = Intents().all()
intents.guild_presences = False

intents = Intents().default()
intents.guild_presences = True

intents = Intents()
intents.guild_prescenes = True

# etc...
```

File: corded/client.py [hunk 1]
```diff
@@ -44,12 +45,12 @@ def __init__(
 
         Args:
             token (str): The token to use for API requests and connecting.
-            intents (int): The intents to use while connecting to the gateway.
+            intents (Intents/int): The intents to use while connecting to the gateway.
```
Comment (vcokltfre) [inline corded/client.py]:
```suggestion
            intents (Union[Intents, int]): The intents to use while connecting to the gateway.
```

File: corded/objects/gateway.py [hunk 2]
```diff
@@ -24,3 +24,90 @@ def typed_data(self):
...
+    @classmethod
+    def all(cls):
+        """A classmethod that will enable all intents including privileged ones
+
+        Returns:
+            Intents: The Intents instance that was created
+        """
+        intents = cls.__new__(cls)
+        for flag in cls.valid.keys():
+            cls.__setattr__(intents, flag, True)
```
Comment (vcokltfre) [inline corded/objects/gateway.py:97]: Did you mean to use `intents.__setattr__(flag, True)`?

File: corded/objects/gateway.py [hunk 3]
```diff
@@ -24,3 +24,90 @@ def typed_data(self):
...
+    @classmethod
+    def default(cls):
+        """A classmethod that will enable all intents excluding privileged ones
+
+        Returns:
+            Intents: The Intents instance that was created
+        """
+        intents = cls.__new__(cls)
+        for flag in cls.valid.keys():
+            cls.__setattr__(intents, flag, True)
```
Comment (vcokltfre) [inline corded/objects/gateway.py:110]: Same as above comment.

**Review by vcokltfre — CHANGES_REQUESTED** [review]

File: corded/objects/gateway.py (hunk 3 above)
Comment (an-dyy) [reply corded/objects/gateway.py:110]: I need to do it like this since `intents.__setattr__` will not work

**Review by vcokltfre — APPROVED** [review]

**Pull request merged**